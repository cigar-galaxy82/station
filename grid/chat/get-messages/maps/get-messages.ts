/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

export const getMessagesTest = (
  provider: string,
  destination: string[],
  options?: RecordingProcessOptions
): void => {
  describe(`chat/get-messages/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('GetMessages', () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: 'chat/get-messages',
          provider,
          useCase: 'GetMessages',
        });
      });

      describe('when specified destination does exist', () => {
        it('performs correctly', async () => {
          const page1 = await superface.run(
            {
              input: {
                destination: destination[0],
                limit: 3,
              },
            },
            options
          );

          expect(page1.isOk).toBeTruthy()
          expect(page1).toMatchSnapshot();

          const page2 = await superface.run(
            {
              input: {
                destination: destination[0],
                limit: 3,
                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
                page: page1.isOk() ? (page1.value as any).nextPage : undefined
              },
            },
            options
          );

          expect(page2.isOk).toBeTruthy()
          expect(page2).toMatchSnapshot();
        });
      });

      describe('when specified destination does not exist', () => {
        it('returns error', async () => {
          await expect(
            superface.run(
              {
                input: {
                  destination: destination[1],
                },
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
