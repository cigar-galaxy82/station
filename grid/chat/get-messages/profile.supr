name = "chat/get-messages"
version = "1.0.0"

"""
GetMessages usecase
"""
usecase GetMessages {
	input {
		destination! string!
		afterDate string
        limit number
		page string
	}
	
	result Messages
	error Error
}

model Message {
	id! string!
	author! string!
	channelId string!
	threadId string

	text string
	createdAt string
	updatedAt string

	reactions [
		{
			emoji string
			count number
            users [string]
		}
	]
}

model Messages {
	previousPage string
	nextPage string
	messages [Message]
}

model Error {
	title!
	detail
}
