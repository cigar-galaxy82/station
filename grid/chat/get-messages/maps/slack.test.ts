import { getMessagesTest } from "./get-messages";

getMessagesTest('slack', 'CF3H7S63W')