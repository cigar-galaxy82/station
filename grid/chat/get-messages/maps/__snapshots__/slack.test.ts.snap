// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/get-messages/slack GetMessages when specified destination does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "messages": Array [
      Object {
        "author": "UF4PRBHL7",
        "createdAt": 1643835555.648069,
        "id": "1643835555.648069",
        "text": "test",
      },
      Object {
        "author": "UF4PRBHL7",
        "createdAt": 1643835553.658259,
        "id": "1643835553.658259",
        "reactions": Array [
          Object {
            "count": 1,
            "emoji": "eyes",
            "users": Array [
              "UF4PRBHL7",
            ],
          },
        ],
        "text": "test",
      },
      Object {
        "author": "UF4PRBHL7",
        "createdAt": 1643798649.182019,
        "id": "1643798649.182019",
        "text": "test :test_tube:",
        "updatedAt": 1643836476,
      },
    ],
    "nextPage": "bmV4dF90czoxNjQzMjE2NDQzMjM0MDc5",
  },
}
`;

exports[`chat/get-messages/slack GetMessages when specified destination does exist performs correctly 2`] = `
Ok {
  "value": Object {
    "messages": Array [
      Object {
        "author": "UF4PRBHL7",
        "createdAt": 1643216443.234079,
        "id": "1643216443.234079",
        "reactions": Array [
          Object {
            "count": 1,
            "emoji": "tada",
            "users": Array [
              "UF4PRBHL7",
            ],
          },
        ],
        "text": "test",
        "updatedAt": 1643285583,
      },
    ],
  },
}
`;

exports[`chat/get-messages/slack GetMessages when specified destination does not exist returns error 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"channel_not_found\\"
}
AST Path: definitions[0].statements[0].responseHandlers[0].statements[0]
Original Map Location: Line 24, column 4",
}
`;
